app/(tabs)/logger.tsx

```diff
@@ -1,88 +1,91 @@
-import { Text, View, StyleSheet, FlatList } from 'react-native';
-import React from 'react';
-import { useFood } from '../FoodProvider';
+import { Text, View, StyleSheet, FlatList } from "react-native";
+import React from "react";
+import { useFood } from "../FoodProvider";
 
 // display somewhere in this page the data input into the addfood page
 // display the data input into the addfood page in a list format
 
 export default function LogScreen() {
-    const { foods, loading } = useFood();
+  const { foods, loading } = useFood();
 
-    if(loading){
-        return (
-            <View style={styles.container}>
-                <Text>Loading...</Text>
-            </View>
-        );
-    }
-
-    const renderMealList = (mealType: string) => (
-        <FlatList
-            data={foods.filter(food => food.mealType === mealType)}
-            keyExtractor={item => item.id}
-            renderItem={({ item }) => (
-                <View style={styles.listItem}>
-                    <Text>{item.name}</Text>
-                    <Text>{item.amount}</Text>
-                </View>
-            )}
-            ListEmptyComponent={<Text style={styles.emptyText}>No foods logged</Text>}
-        />
-        )
+  const safeFoods = Array.isArray(foods) ? foods : [];
```

Comment:
[nit] if the `useFoods` hook is returning a `FoodItem[]`, can't we always assume that `foods` will be an array? Do we need this check here? 🤔